src/mbgl/style/source_impl.cpp

```diff
@@ -207,16 +207,23 @@ static Point<int16_t> coordinateToTilePoint(const UnwrappedTileID& tileID, const
 }
 
 std::unordered_map<std::string, std::vector<Feature>> Source::Impl::queryRenderedFeatures(const QueryParameters& parameters) const {
+    std::unordered_map<std::string, std::vector<Feature>> result;
+    if (renderTiles.empty()) {
```

Comment:
The algorithm works fine without checking these boundary conditions right? Why increase the amount of code?